src/GlobalExceptionHandler/WebApi/ExceptionHandlerConfiguration.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+
+namespace GlobalExceptionHandler.WebApi
+{
+	public class ExceptionHandlerConfiguration
+	{
+		readonly IDictionary<Type, ExceptionConfig> _exceptionConfiguration = new Dictionary<Type, ExceptionConfig>();
+
+		Type[] _exceptionConfgurationTypesSortedByDepthDescending;
+
+		public ExceptionHandlerConfiguration(Func<Exception, HttpContext, Task> defaultFormatter) => DefaultFormatter = defaultFormatter;
+
+		public IHasStatusCode ForException<T>() where T : Exception
+		{
+			var type = typeof(T);
+			return new ExceptionRuleCreator(_exceptionConfiguration, type);
+		}
+
+		public void UseDefaultMessageFormatter(Func<Exception, HttpContext, Task> formatter)
+		{
+			DefaultFormatter = formatter;
+		}
+
+		internal IDictionary<Type, ExceptionConfig> ExceptionConfiguration => _exceptionConfiguration;
+
+		internal Func<Exception, HttpContext, Task> DefaultFormatter { get; private set; }
+
+		internal RequestDelegate BuildHandler()
+		{
+			_exceptionConfgurationTypesSortedByDepthDescending = _exceptionConfiguration.Keys
+				.OrderBy(x => x, new ExceptionTypePolymorphicComparer())
```

Comment:
We should use `.OrderByDescending` here instead, and then make the `ExceptionTypePolymorphicComparer` do `'return depthOfX.CompareTo(depthOfY);` instead of `return depthOfY.CompareTo(depthOfX);`

Reply:
Good spot, I'll update that 👍 

I've got a branch where I'm building in a few changes based off of this, hoping to get it merged soon. Went with `app.UseExceptionHandler().WithConventions(x =>{});` in the end for the vanity extension point.